functions/export-csv.js

```diff
@@ -15,79 +15,84 @@
  * limitations under the License.
  */
 
-'use strict';
+"use strict";
 
-const {db} = require('./common/context');
+const csv = require("@fast-csv/format");
+const { db } = require("./common/context");
 
-function exportCsv(req, res) {
```

Comment:
Feature objects don't actually contain observations, since features and observations are stored in separate collections.

Reply:
Out of curiosity: why are they stored in different collections? The observations at least have a feature key, no?

Then something like:
features = db.fetchFeatures(projectId, layerId);
for (feature : features) {
observations = db.fetchObservations(projectId, layerId, featureId)
for (observation : observations) {
csv[feature].add(observation);
}
}

The data model is one project can have multiple layers, one layer can have multiple features and each feature can have multiple observations, right?

Reply:
It has to do with how Firestore provides access to data. We can fetch and filter all docs in a single collection, but not across multiple collections. We originally had observations as docs inside feature, but that forced us to do many queries (one per feature) to get all observations. We flattened to hierarchy to allow observations to be fetched with a single query.

Reply:
Still not very convinced of why the above suggested solution does not work, but I think resolving that needs a meeting sync. Let me unblock you for now.

Reply:
If we nest observations in features in the db it will require n queries (n=# of features) to fetch all observations; unfortunately there's no way in Firebase to do a recursive query. Happy to discuss more offline!